hamster_gtk/preferences/widgets/time_entry.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+
+# This file is part of 'hamster-gtk'.
+#
+# 'hamster-gtk' is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# 'hamster-gtk' is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with 'hamster-gtk'.  If not, see <http://www.gnu.org/licenses/>.
+
+"""This module provides a widget for entering time."""
+
+from __future__ import absolute_import
+
+import datetime
+
+from gi.repository import Gtk
+from six import text_type
+
+# [FIXME]
+# Remove once hamster-lib has been patched
+from hamster_gtk.helpers import _u
+
+from .config_widget import ConfigWidget
+
+
+class TimeEntry(Gtk.Entry, ConfigWidget):
+    """A widget for entering time."""
+
+    # Required else you would need to specify the full module name in ui file
+    __gtype_name__ = 'TimeEntry'
+
+    def get_config_value(self):
+        """
+        Return selected time.
```

Comment:
I agree with the best case solution to be input validation as well as with that this should be a separate issue.
However, the docstring should document the actual behavior as is. And that would be raising an ``ValueError``? exception if not matching one of the two formats.